packages/chat-component-bindings/src/handlers/createHandlers.ts

```diff
@@ -81,28 +82,36 @@ export const createDefaultChatHandlers = memoizeOne(
         let remainingMessagesToGet = messagesToLoad;
         let isAllChatMessagesLoaded = false;
         let earliestTime = Number.MAX_SAFE_INTEGER;
-        while (remainingMessagesToGet >= 1) {
-          const message = await messageIterator.next();
-          if (message?.value?.id) {
-            if (parseInt(message.value.id) < earliestTime) {
-              earliestTime = parseInt(message.value.id);
+        try {
+          while (remainingMessagesToGet >= 1) {
+            const message = await messageIterator.next();
+            if (message?.value?.id) {
+              if (parseInt(message.value.id) < earliestTime) {
+                earliestTime = parseInt(message.value.id);
+              }
             }
-          }
 
-          if (message.value?.type && message.value.type === 'text') {
-            remainingMessagesToGet--;
-          }
+            if (message.value?.type && message.value.type === 'text') {
+              remainingMessagesToGet--;
+            }
 
-          // We have traversed all messages in this thread
-          if (message.done) {
-            isAllChatMessagesLoaded = true;
-            break;
+            // We have traversed all messages in this thread
+            if (message.done) {
+              isAllChatMessagesLoaded = true;
+              break;
+            }
           }
+        } catch (e) {
+          console.log(e);
         }
         // keep fetching read receipts until read receipt time < earlist message time
-        let readReceipt = await readReceiptIterator.next();
-        while (!readReceipt.done && parseInt(readReceipt?.value?.chatMessageId) >= earliestTime) {
-          readReceipt = await readReceiptIterator.next();
+        try {
+          let readReceipt = await readReceiptIterator.next();
+          while (!readReceipt.done && parseInt(readReceipt?.value?.chatMessageId) >= earliestTime) {
+            readReceipt = await readReceiptIterator.next();
+          }
+        } catch (e) {
+          console.log(e);
```

Comment:
hesitant around console.log in library code. 

Can we throw some kind of exception instead?

Reply:
this is how we catch the other errors coming out of the chat thread in the adapter we do this for `getProperties` and `listParticipants` in `AzureCommunicationsChatAdapter.ts` should we throw the errors here and then catch them in the adapter as well?

Reply:
The handler layer is more like for those users who is building on top of component+selector level, swallowing the error and not throwing it sounds a little dangerous and it's actually a behavior change.
If this is just to stop the iteration, then the best way is just to stop the iteration/loop then throw the error following previous behavior

Reply:
Will make that change!